### DISCUSSION TITLE
Is Open Watcom alive?

### DISCUSSION THREAD

**Discussion by ta0590897** [open-watcom/open-watcom-v2#101] [General]
Is the OW project alive in any flavor or fork? I don't understand where to ask questions. Sorry for spamming "Issues" but I don't understand where to ask a question.

**Comment by ArmstrongJ** [comment]
It sure is alive!  Work continues on it at a surprisingly brisk pace considering the limited number of contributors.  Looking at the number of commits and how recent they are, I'd say the project is still alive.  I've submitted two pull requests this week, in fact.

I don't think we have a good place to ask questions.  Anyone have any ideas outside of GitHub issues?  There was some interest in using a [subreddit](http://www.reddit.com/r/OpenWatcom/), but that hasn't gained much traction.

**Comment by ta0590897** [comment]
Great! Because from what I've been able to find out, Open Watcom is the only C++ compiler in any sort of active development that still supports the 386 and MS-DOS, and 16-bit x86 development in general.

I think a subreddit is a good idea, actually, _anything_ is a good idea, because currently it has been quite a hassle to get open watcom up and running, and if I didn't have a particular need for the 386 compilation, I would have given up a long time ago. Really, the fact that the wiki says it was last updated in 2010 will probably turn away most people.

Though news groups and mailing lists are perhaps not the best option; they are a bit archaic. Still better than nothing, and you can't beat the "backwards compatibility factor" of USENET.

I'll post my question in the subreddit it see what happens.

**Comment by ArmstrongJ** [comment]
There's a very good chance I'll be the one answering as there are only two subscribers to that [subreddit](http://www.reddit.com/r/OpenWatcom/), and we're both mods.

**Comment by pchapin** [comment]
The main Open Watcom site seems inactive but I admit I haven't checked it recently. Essentially what has happened is that Jiri, who is amazingly productive, decided to create this fork for various reasons I believe he's documented elsewhere. To my mind it is this fork where the "action" is and where I have been putting my recent (microscopic) contributions.

As far as asking questions go I agree that is a problem as using this space is a bit of an abuse. I don't have a good suggestion... maybe the forums on SourceForge?

**Comment by ta0590897** [comment]
Agree about the github abuse, in principle, but otoh there is almost zero activity so what's the harm?

Would be worse if all of the effort put in to OW by various people went to waste because nobody could get support...

**Comment by pchapin** [comment]
I agree that using GitHub for questions isn't the end of the world and there are some advantages. Maybe we should create a tag for "question" or some such to go along with the predefined ones. I believe GitHub does allow projects to define their own issue tags.

**Comment by ghost** [comment]
Maybe replace the readme.txt that is shown in the repository with something that explains what this fork is all about, who made it, etc? Because right now if someone visits this repository he'll think that http://openwatcom.org/ is still this project's page. And TBH the readme shown isn't very helpful for someone trying to figure out what this repository is all about.

**Comment by ArmstrongJ** [comment]
I like @badsector's idea.  A lot of what's contained in readme.txt could be moved to, for example, building.txt or something similar.  The readme should explain the project and reference another text file for build instructions.  Maybe the explanation of the directories should stay, though, since Open Watcom's directory structure is so unlike any other project.

Does anyone want to work on this, or should I take a crack at it?

**Comment by pchapin** [comment]
I was actually thinking about converting the readme to Markdown format so it looks nicer when viewed through GitHub. If people agree, why don't I do that and make the change talked about here as well. If I don't get to it this weekend, then someone else can step in. How does that sound?

**Comment by ArmstrongJ** [comment]
@pchapin sounds like a great plan! :+1:

**Comment by jmalak** [comment]
Peter, it will be fine if you do this.
I am sorry, but I am not big writer and also my English is bad.
I am able to help with some details about OW if it will be necessary.

**Comment by pchapin** [comment]
I modified the README (now in Markdown format) and created a separate BUILD.md with build instructions. I also reorganized the information a bit and edited the English. I don't think the new files are necessarily the last word on what we want but I believe they are a step forward. Hopefully they will serve as a good starting point for further enhancement.

**Comment by jmalak** [comment]
Great. Thanks.

**Comment by ytrezq** [comment]
@ta0590897 yes it is alive again. Source code development on Perforce [restarted](http://perforce.openwatcom.org:4000/@md=d&cd=//depot/&c=uBY@//depot/?ac=43&mx=50).

**Comment by ArmstrongJ** [comment]
@ytrezq the Perforce server is up, but this fork is drastically further along in regards to the compiler and standard libraries.  The commits appear to be some work towards reverse engineering Open Watcom's `wgml.exe`, which is also a commendable enterprise if it ever reaches feature parity.

**Comment by ytrezq** [comment]
@ArmstrongJ so among the recents bugs fix [during the last week](http://perforce.openwatcom.org:4000/@md=d&cd=//depot/&c=uBY@//depot/?ac=43&mx=50), is there really none which have not been corrected since the time of the fork ?

**Comment by ArmstrongJ** [comment]
@ytrezq oh no, that wasn't what I was trying to say.  The commits, like I said, were moving towards implementing a new version of wgml (I think, don't quote me).  Those fixes haven't been pulled into this repo to the best of my knowledge.  The Perforce Open Watcom development is taking a different track, though.

Jiri and developers here have been working on adding support for AMD64, better C99 compliance, additional Windows API updates, and, lately, ARM as a host architecture.  Those changes are not being pulled into the Perforce repo.  The two are effectively separate projects.

**Comment by ytrezq** [comment]
My point was if we can pull intersting changes from Perforce into this fork in the future.

**Comment by jmalak** [comment]
Yes it is possible by git cherry-pick function. It is one of reasons why owp4v1copy repository exists.
But I didn't do it for long time, because no real OW development exists in Perforce.

**Comment by sparky4** [comment]
oh!!! i have bugs to report ^^ and i have no idea how to compile it! ^^;

**Comment by jmalak** [comment]
Basic description of build process and commands are here https://github.com/open-watcom/open-watcom-v2/wiki/Build . If there is something unclear simply ask question, somebody help you. 
You can also create issue here as bug report with problem description.

**Comment by sparky4** [comment]
Thank you! sorry for late reply~

**Comment by jmalak** [comment]
No problem.
